### PR TITLE
servicestate: use systemctl enable+start and disable+stop instead of --now flag

### PR DESCRIPTION
**Pull request by stolowski**
The `--now` flag we pass to `systemctl enable/disable` commands with `snap start/stop..` or `snapctl start/stop ...` is not avalable on systemd version shipped with 14.04 and causes an error.
This PR removes the use of `--now` and executes enable, followed by start or disable followed by stop instead. These are alternatives to `--now` suggested by man systemctl.

### CODE REVIEW

File: overlord/servicestate/servicestate.go [hunk 1]
```diff
@@ -44,68 +44,83 @@ type ServiceActionConflictError struct{ error }
...
-					return chg.ID() != otherTask.Change().ID()
+	for _, cmd := range ctlcmds {
+		// the argv to call systemctl will need at most one entry per appInfo,
+		// plus one for "systemctl", one for the action, and sometimes one for
+		// an option. That's a maximum of 3+len(appInfos).
+		argv := make([]string, 2, 3+len(appInfos))
+		argv[0] = "systemctl"
+		argv[1] = cmd
+
+		snapNames := make([]string, 0, len(appInfos))
```
Comment (chipaca) [inline overlord/servicestate/servicestate.go]: why're you doing this in a loop, with the lock held?

File: overlord/servicestate/servicestate.go [hunk 2]
```diff
@@ -44,68 +44,83 @@ type ServiceActionConflictError struct{ error }
...
+						// if same change, then return false (no conflict)
+						return chg.ID() != otherTask.Change().ID()
+					}
+					return true
 				}
-				return true
 			}
 		}
+
+		if err := snapstate.CheckChangeConflictMany(st, snapNames, checkConflict); err != nil {
```
Comment (chipaca) [inline overlord/servicestate/servicestate.go]: I suspect you don't need to check for conflicts in a loop, either, but I could be wrong on that

Comment (stolowski) [reply overlord/servicestate/servicestate.go]: You're totally right in your both comments regarding doing all this in the loop, this is wasteful. I skimmed through this code and missed that, thanks for catching.

**Comment by lichtamberg** [comment]
Would be great if you could release a new version with this patch included, since our system can't upgrade some snaps due to this issue ❤️

**Comment by lichtamberg** [comment]
@stolowski Could you release a new version for Ubuntu 14.04? Our system cant upgrade some snaps due to this issue?

**Comment by stolowski** [comment]
@lichtamberg I'm sorry for not responding to you earlier request, I missed it.
We release new core snap updates on regular basis, every few weeks, this makes the new version of snapd available to all supported systems. The new version that will include this change should be out very soon.

**Pull request merged**